examples/notebooks/beam-ml/alloydb_product_catalog_embeddings.ipynb

```diff
@@ -2148,6 +2160,599 @@
+        "    )\n",
+        "\n",
+        "# Set up pipeline options\n",
+        "options = PipelineOptions()\n",
+        "options.view_as(StandardOptions).streaming = True  # This makes it a streaming pipeline\n",
+        "\n",
+        "# Create and run the pipeline\n",
+        "with beam.Pipeline(options=options) as p:\n",
+        "    _ = (\n",
+        "        p\n",
+        "        # SOURCE: Read continuously from PubSub\n",
+        "        | \"Read from PubSub\" >> beam.io.ReadFromPubSub(\n",
+        "            subscription=\"projects/my-project/subscriptions/my-subscription\"\n",
+        "        )\n",
+        "        \n",
+        "        # WINDOWING: Group messages into 10-second windows\n",
+        "        | \"Window\" >> beam.WindowInto(FixedWindows(10))\n",
+        "        \n",
+        "        # TRANSFORM: Parse JSON messages into Chunks\n",
+        "        | \"Parse Messages\" >> beam.Map(parse_message)\n",
+        "        \n",
+        "        # ML PROCESSING: Generate embeddings with Huggingface\n",
+        "        | \"Generate Embeddings\" >> MLTransform(write_artifact_location=tempfile.mkdtemp())\n",
+        "            .with_transform(HuggingfaceTextEmbeddings(\n",
+        "                model_name=\"sentence-transformers/all-MiniLM-L6-v2\"\n",
+        "            ))\n",
+        "        \n",
+        "        # SINK: Write to AlloyDB vector database\n",
+        "        | \"Write to AlloyDB\" >> VectorDatabaseWriteTransform(\n",
+        "            AlloyDBVectorWriterConfig(\n",
+        "                connection_config=AlloyDBConnectionConfig(\n",
+        "                    jdbc_url=\"jdbc:postgresql://host:port/database\",\n",
+        "                    username=\"user\",\n",
+        "                    password=\"password\"\n",
+        "                ),\n",
+        "                table_name=\"product_embeddings\",\n",
+        "                conflict_resolution=ConflictResolution(\n",
+        "                    on_conflict_fields=\"id\",\n",
+        "                    action=\"UPDATE\",\n",
+        "                    update_fields=[\"embedding\", \"content\", \"metadata\"]\n",
+        "                )\n",
+        "            )\n",
+        "        )\n",
+        "    )\n",
+        "```"
+      ],
+      "metadata": {
+        "id": "hWEpljFpkvp3"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Save our Pipeline to a python file\n",
+        "\n",
+        "To launch our pipeline job on DataFlow we save our pipeline code to a local file `streaming_ingestion_pipeline.py`.\n",
+        "\n",
+        "Expand the hidden cell to see the entire pipeline content."
+      ],
+      "metadata": {
+        "id": "0PBGyr6mnSLs"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "#@title Save pipeline to streaming_ingestion_pipeline.py\n",
+        "file_content = \"\"\"\n",
```

Comment:
Rather than saving this to a file and running it that way, can we just run it directly through colab like https://github.com/apache/beam/blob/8c588f2d5c53f637eb745478beead92e4c5e19ce/examples/notebooks/beam-ml/automatic_model_refresh.ipynb ?